### PR TITLE
Update progress script with new assets categories and update csv output format

### PR DESCRIPTION
**Pull request by AngheloAlf**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

- Adds the new asset categories we had discussed, reorganizing the csv files and the XMLs as needed.
  - The asset categories are: `archive`, `audio`, `deleted`, `interface`, `misc`, `objects`, `scenes`, `segment` and `text`
  - Should be noted if an asset is not in the correct folder, it won't be counted in the progress. To know which file should go in each folder, refer to the csvs in `tools/filelists/mm.us.rev1/`
- Changed the csv format outputted by the progress script to accommodate the new format the new website will be using.
- Automatically generated names in assets will be filtered out when calculating the "Matching" progress.

### CODE REVIEW

**Review by EllipticEllipsis — COMMENTED** [review]
This looks really good overall, looking forward to getting a nice-looking graph with recategorised assets. I have a few comments that are mostly about naming things.

One more thing: this reads the mapfile now, at least for assets? How does this relate to what it does for functions? I know it was much slower than other projects' before... from what I'm reading in the lines around 50 or so, it still reads the asm files? Could we switch to reading the mapfile for functions too? (Not recommending doing this now necessarily: this PR's changing a lot about the script already, but now that it needs the mapfile, it really ought to use it for everything, especially if it would be a significant speed improvement.) (The alternative, of course is to use `wc -c` or equivalent on the `baserom` folder and add those up)

File: tools/filelists/mm.us.rev1/misc.csv [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
+1127,week_static
+1128,daytelop_static
+1129,ger_daytelop_static
+1130,fra_daytelop_static
+1131,esp_daytelop_static
+1132,d2_fine_static
+1133,d2_cloud_static
+1134,d2_fine_pal_static
 1135,elf_message_field
 1136,elf_message_ydan
```
[the remark this answers is not in the document]
Comment (AngheloAlf) [reply tools/filelists/mm.us.rev1/misc.csv:25]: Which csv category do you propose?
OoT have them in `src/elf_message/`, so `asm.csv`?

File: tools/progress.py [hunk 2]
```diff
@@ -273,16 +344,21 @@ def GetRemovableSize(functions_to_count):
...
-
-    print(",".join(csv_list))
+    csv_list = [
+        version, timestamp, git_hash, src, total,
+        boot, boot_size, code, code_size, ovl, ovl_size,
+    ]
+    csv_list += [
+        assets, assets_total,
+    ]
+    for assetCat in assetsCategories:
```
[the remark this answers is not in the document]
Comment (AngheloAlf) [reply tools/progress.py:358]: I would prefer to add a comment in the `assetsCategories` saying the order should not be changed to not break the website

[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply tools/progress.py:358]: That would work too.

File: tools/filelists/mm.us.rev1/misc.csv (hunk 1 above)
[the remark this answers is not in the document]
Comment (EllipticEllipsis) [reply tools/filelists/mm.us.rev1/misc.csv:25]: I suppose, since they would be disassembled by a script, like the asm.

**Comment by AngheloAlf** [comment]
> One more thing: this reads the mapfile now, at least for assets? How does this relate to what it does for functions? I know it was much slower than other projects' before... from what I'm reading in the lines around 50 or so, it still reads the asm files? Could we switch to reading the mapfile for functions too? (Not recommending doing this now necessarily: this PR's changing a lot about the script already, but now that it needs the mapfile, it really ought to use it for everything, especially if it would be a significant speed improvement.) (The alternative, of course is to use wc -c or equivalent on the baserom folder and add those up)

The progress script has always required the map file to do the progress calculations.
Using the map file to get those sizes should be possible, but as you said, it kinda is a bit out of scope of this PR

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Looks good to me! General reminder to resolve conversations as you address code changes.

**Review by EllipticEllipsis — APPROVED** [review]

**Pull request merged**